packages/react-components/react-positioning/src/types.ts

```diff
@@ -1,26 +1,37 @@
-import * as PopperJs from '@popperjs/core';
 import * as React from 'react';
 
+type Rect = {
```

Comment:
Consider if you DOMRect would be a good structure to use here.  It has x, y, width, and height, but also top, left, right, and bottom.  It can be annoying to create instances of a DOMRect because of the fromRect method, but might fit well with positioning.